tests/shared/src/test/scala/scala/meta/tests/parsers/DefnSuite.scala

```diff
@@ -172,6 +172,61 @@ class DefnSuite extends ParseSuite {
     ) = templStat("def f(x: Int): Int = macro impl")
   }
 
+  test("braces-in-functions") {
```

Comment:
how's this treated:
```
if (...) { ... }.select else {...} + {...}
try {...}.select
```
what is `location == BlockStat`?

Reply:
I think `BlockStat` means we are in a block, which can be either indented, `=>` or {}. At least that's what I think.
I will check the other example